src/test_disk.rs

```diff
@@ -1514,11 +1514,60 @@ fn repair_ext(device: &Path) -> BynarResult<()> {
     }
 }
 
+// Run the smartctl checks against the disk if libata fails
+#[cfg_attr(test, mockable)]
+fn run_smartctl_check(device: &Path) -> BynarResult<bool> {
+    // Enable Smart Scan
+    let out = Command::new("smartctl")
```

Comment:
There's actually a library that you can use to do this: https://crates.io/crates/libatasmart.  I wrote it 2yrs ago.  It probably needs some updates